### DISCUSSION TITLE
Try to get the ReadDeviceInformation through the forwarder.

### DISCUSSION THREAD

**Discussion by Fredo70** [pymodbus-dev/pymodbus#1637] [Help] [answered]
Hi
For my project (Raspberry pi between 2 existing devices) I need to get all the "ReadDeviceInformation" requests through the forwarder. As it looks, it is not working at the moment.
I thought, first I have to do this in the RemoteSlaveContext. Maybe derive a separate class from it. But what I have seen is that the ReadDeviceInformationRequest does not take the context into account. So it will not forward the request. Right? I thought I could create my own request based on the ReadDeviceInformationRequest. But how do I get the server to decode my request? Do I have to create my own decoder?
It's such a guessing game that I feel like I'm on the wrong track. I get stuck.

**Comment by janiversen** [comment, marked as answer]
The datastore only deal with values, not the rest of the calls.

I am planning a new forwarder that catches incoming requests at framer level and retransmit the pure request. This is the next level of the transport directory, meaning new: server/client encoder/decoder and framers.

You should be able to do something similar by making a custom framer. Surely there are other solutions, but this is the one I deem most general.
